src/android.ts

```diff
@@ -237,6 +234,162 @@ export class AndroidRobot implements Robot {
 
 		return rotation === "0" ? "portrait" : "landscape";
 	}
+
+	private getParsedXml(): UiAutomatorXml {
+		const dump = this.adb("exec-out", "uiautomator", "dump", "/dev/tty");
+
+		const parser = new xml.XMLParser({
+			ignoreAttributes: false,
+			attributeNamePrefix: ""
+		});
+
+		return parser.parse(dump) as UiAutomatorXml;
+	}
+
+	private getScreenElementRect(node: UiAutomatorXmlNode): ScreenElementRect {
+		const bounds = String(node.bounds);
+
+		const [, left, top, right, bottom] = bounds.match(/^\[(\d+),(\d+)\]\[(\d+),(\d+)\]$/)?.map(Number) || [];
+		return {
+			x: left,
+			y: top,
+			width: right - left,
+			height: bottom - top,
+		};
+	}
+
+	// --- TV Specific Methods ---
+
+	public navigateToItemWithLabel(label: string) {
+		this.requireAndroidTv();
+		let currentDirection = this.getNextDpadDirectionToItemWithLabel(label);
+
+		while (currentDirection) {
+			this.pressDpadInternal(currentDirection);
+			currentDirection = this.getNextDpadDirectionToItemWithLabel(label);
+		}
+	}
+
+	public pressDpad(dpadButton: DpadButton) {
+		this.requireAndroidTv();
+		this.pressDpadInternal(dpadButton);
+	}
+
+	private getNextDpadDirectionToItemWithLabel(label: string): DpadButton | null {
+		const parsedXml = this.getParsedXml();
+		const targetElement = this.findElemenWithLabel(parsedXml.hierarchy.node, label);
+		const focusedElement = this.findFocusedElement(parsedXml.hierarchy.node);
```

Comment:
`memo`
With the latest layout information, I know which is my currently focused element and which direction I need to be heading to in order to reach the target coordinates.

That is the reason why before searching for the focused and target elements, I need to call `getParseXml` to get the refresh the layout information.